File: Swift/rotate-string.swift
class Solution {
    func rotateString(_ A: String, _ B: String) -> Bool {
        guard A != B else {
            return true
        }
        guard A.count == B.count else {
            return false
        }

        var a = A

        for _ in 0..<A.count-1 {
            a.append(a.removeFirst())
            if a == B {
                return true
            }
        }

        return false
    }
}

File: Python/rotate-string.py
class Solution:
    def rotateString(self, A, B):
        """
        :type A: str
        :type B: str
        :rtype: bool
        """
        if len(A) != len(B):
            return False

        if A == B:
            return True
        
        a = list(A)
        b = list(B)

        for _ in range(len(a)-1):
            a.append(a.pop(0))
            if a == b:
                return True

        return False
